src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/RegisterCsrfFeaturesPass.php

```diff
@@ -39,12 +42,24 @@ private function registerCsrfProtectionListener(ContainerBuilder $container): vo
             return;
         }
 
+        if ($container::willBeAvailable('symfony/expression-language', ExpressionLanguage::class, ['symfony/security-bundle'])) {
```

Comment:
I'd rather wire this part exactly the same way we do it for `cache.security_is_granted_attribute_expression_language`: define the service in the config file and remove it not when the component is missing, but when the required service is missing.

Reply:
@nicolas-grekas are the changes made in [this fixup commit](https://github.com/symfony/symfony/pull/54443/commits/58a2140da3a6a638d3764f65bd823f4a4b8d47b9) what you expect?
what about the `controller.is_csrf_token_valid_attribute_listener` service introduced in #52961? is it OK to keep it registered in `RegisterCsrfFeaturesPass` as it depends on the `security.csrf.token_manager` service? or should I move also?